src/plugins/intel_gna/src/frontend/model_quantizer.hpp

```diff
@@ -148,13 +148,28 @@ class ModelQuantizer {
                 inf_loop_count++;
             } else {
                 if (inf_loop_count > 0 &&
-                    (inf_loop_history.size() % inf_loop_pattern.size() == 0 || sf.allLayersProcessed()) &&
```

Comment:
if we decide to add only extra condition check to prevent crash "(inf_loop_history.size() > inf_loop_pattern.size())" 
we can see exception with infinite loop.